pkg/apis/kops/v1alpha2/instancegroup.go

```diff
@@ -165,6 +165,15 @@ type InstanceGroupSpec struct {
 	// MaxInstanceLifetime to the maximum amount of time, in seconds, that an instance can be in service.
 	// Value expected must be in form of duration ("ms", "s", "m", "h")
 	MaxInstanceLifetime *metav1.Duration `json:"maxInstanceLifetime,omitempty"`
+	// Preemptible: Defines whether the instance is preemptible.
+	Preemptible *bool `json:"preemptible,omitempty"`
+	// ProvisioningModel: Specifies the provisioning model of the instance.
+	//
+	// Possible values:
+	//   "SPOT" - Heavily discounted, no guaranteed runtime.
+	//   "STANDARD" - Standard provisioning with user controlled runtime, no
+	// discounts.
```

Comment:
Let's use the same format as with `InstanceMetadata`. We can allow both uppercase and lowercase spelling for values.
```suggestion
	// ProvisioningModel: Specifies the provisioning model of the instance.
	// Valid values:
	//   'standard: (default) standard provisioning with user controlled run time, no discounts
	//   'spot': heavily discounted, no guaranteed run time.
```

Reply:
Is there any way of shoehorning this into the AWS model? It would be nice to avoid having concepts that are fairly similar across clouds different in our models.

Reply:
There is `maxPrice` field with decimal type to enable spot mode in aws. I think it is very uncompatible with gce approach with enum type

Reply:
fixed

Reply:
There is also these settings.
https://pkg.go.dev/k8s.io/kops/pkg/apis/kops?utm_source=godoc#MixedInstancesPolicySpec

Would be awkward, but you could say that GCP only allows 0 or 100 for `OnDemandAboveBase`

Reply:
After some discussions, the new name should be `GCPProvisioningModel`. Plase update the PR.

Reply:
Fixed